src/journal/journald-syslog.c

```diff
@@ -124,11 +124,13 @@ static void forward_syslog_raw(Server *s, int priority, const char *buffer, cons
         if (LOG_PRI(priority) > s->max_level_syslog)
                 return;
 
-        IOVEC_SET_STRING(iovec, buffer);
+        iovec.iov_base = (char *) buffer;
+        iovec.iov_len = buffer_len;
+
         forward_syslog_iovec(s, &iovec, 1, ucred, tv);
 }
 
-void server_forward_syslog(Server *s, int priority, const char *identifier, const char *message, const struct ucred *ucred, const struct timeval *tv) {
+void server_forward_syslog(Server *s, int priority, const char *identifier, const char *message, size_t message_len, const struct ucred *ucred, const struct timeval *tv) {
```

Comment:
IIRC, variable was used in the past because my original implementation didn't trim whitespace from native messages.